joint_trajectory_controller/test/test_trajectory_actions.cpp

```diff
@@ -250,108 +301,80 @@ TEST_F(TestTrajectoryActions, test_success_multi_point_sendgoal) {
-      rclcpp::Duration wait = rclcpp::Duration::from_seconds(3.0);
-      auto end_time = start_time + wait;
-      while (rclcpp::Clock().now() < end_time) {
-        test_robot->read();
-        traj_controller->update();
-        test_robot->write();
-      }
-    };
-  std::thread controller_hw_thread(thread_func);
-
-  // common_goal_response and common_result_response
-  // sometimes doesnt receive calls when we dont sleep
-  std::this_thread::sleep_for(std::chrono::milliseconds(300));
+  traj_lifecycle_node_->declare_parameter("constraints.joint1.goal", 0.0);
+  traj_lifecycle_node_->declare_parameter("constraints.joint2.goal", 0.0);
+  traj_lifecycle_node_->declare_parameter("constraints.joint3.goal", 0.0);
+  traj_lifecycle_node_->set_parameter(rclcpp::Parameter("constraints.joint1.goal", 0.1));
+  traj_lifecycle_node_->set_parameter(rclcpp::Parameter("constraints.joint2.goal", 0.1));
+  traj_lifecycle_node_->set_parameter(rclcpp::Parameter("constraints.joint3.goal", 0.1));
 
-  GoalOptions opt;
-  opt.goal_response_callback =
-    std::bind(&TestTrajectoryActions::common_goal_response, this, _1);
-  opt.result_callback =
-    std::bind(&TestTrajectoryActions::common_result_response, this, _1);
-  opt.feedback_callback = nullptr;
+  SetUpExecutor();
+  SetUpControllerHardware();
 
   // send goal
   {
     std::vector<JointTrajectoryPoint> points;
     JointTrajectoryPoint point;
     point.time_from_start = rclcpp::Duration::from_seconds(0.0);  // start asap
-    point.positions.resize(joint_names.size());
+    point.positions.resize(joint_names_.size());
 
     point.positions[0] = 1.0;
     point.positions[1] = 2.0;
     point.positions[2] = 3.0;
     points.push_back(point);
 
-    sendActionGoal(points, 1.0, opt);
+    sendActionGoal(points, 1.0, goal_options_);
   }
-  controller_hw_thread.join();
+  controller_hw_thread_.join();
 
-  EXPECT_EQ(true, common_goal_accepted);
-  EXPECT_EQ(rclcpp_action::ResultCode::SUCCEEDED, common_resultcode);
+  EXPECT_EQ(true, common_goal_accepted_);
+  EXPECT_EQ(rclcpp_action::ResultCode::SUCCEEDED, common_resultcode_);
   EXPECT_EQ(
     control_msgs::action::FollowJointTrajectory_Result::SUCCESSFUL,
-    common_action_result_code);
+    common_action_result_code_);
 
-  EXPECT_NEAR(1.0, test_robot->pos1, common_threshold);
-  EXPECT_NEAR(2.0, test_robot->pos2, common_threshold);
-  EXPECT_NEAR(3.0, test_robot->pos3, common_threshold);
+  EXPECT_NEAR(1.0, test_robot_->pos1, COMMON_THRESHOLD);
+  EXPECT_NEAR(2.0, test_robot_->pos2, COMMON_THRESHOLD);
+  EXPECT_NEAR(3.0, test_robot_->pos3, COMMON_THRESHOLD);
+}
 
-  // start again
-  common_goal_accepted = false;
-  common_resultcode = rclcpp_action::ResultCode::UNKNOWN;
-  controller_hw_thread = std::thread(thread_func);
+TEST_F(TestTrajectoryActions, test_goal_tolerances_multi_point_success) {
+  // set tolerance parameters
+  traj_lifecycle_node_->declare_parameter("constraints.joint1.goal", 0.0);
```

Comment:
`declare_parameter` and then `set_parameter` are both called but we could directly set the values in `declare_parameter`!
http://docs.ros2.org/foxy/api/rclcpp/classrclcpp_1_1Node.html#a095ea977b26e7464d9371efea5f36c42

Reply:
Done.